tests/System.IO.Pipelines.Tests/BackpressureTests.cs

```diff
@@ -170,5 +170,16 @@ public void FlushAsyncAwaitableResetsOnCommit()
 
             Assert.False(flushAsync.IsCompleted);
         }
+
+        [Fact]
+        public void ThrowsOnAllocDuringFlush()
```

Comment:
I just realized this will make the stream code impossible to write... https://github.com/aspnet/KestrelHttpServer/blob/751a0e2e7e92fc2adc23092aeec5117a2afe5a1b/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketOutput.cs#L83. We can't do this one.